### PR TITLE
Doc update; new example; new tutorial for artifact dependencies

### LINKED ISSUE #4967: Document Artifact dependency feature for dockerfile artifacts.
We announcing the artifact dependency support in release notes , but just realized the docs are missing on how to use this feature. 

Lets add a tutorial or an explanation on how to configure this for dockerfile artifacts.
- explain the example. 
- Add a new feature here in this [matrix](https://skaffold.dev/docs/references/deprecation/#skaffold-features). 
   ( Reach out to @tejal29 on the excel which drives this table. 
- Update release notes and change log for v1.16.0 to add the docs.

### PR DESCRIPTION
**Pull request by gsquared94**
Fixes #4967, Related #4713 

- [Website] Updates the Docker builder docs with details about artifact dependency resolution as build-args
- Add new example `simple-artifact-dependency`
- [Website] Add tutorial about artifact dependency feature using `simple-artifact-dependency`

### CODE REVIEW

**Comment by gsquared94** [comment]
Specifically: http://34.94.209.155:1313/docs/tutorials/artifact-dependencies/

**Comment by MarlonGamez** [comment]
This looks good! I would maybe like to maybe see what the output looks like when running skaffold build at the end of the page, not sure if others feel the same?

File: docs/content/en/docs/tutorials/_index.md [hunk 1]
```diff
@@ -26,3 +26,4 @@ Take a look at our other guides:
 | Detailed Tutorials |
 |--------------------|
 | [Custom Build Script]({{< relref "/docs/tutorials/custom-builder" >}}) |
+| [Inter-Artifact Dependencies]({{< relref "/docs/tutorials/artifact-dependencies" >}}) |
```
Comment (tejal29) [inline docs/content/en/docs/tutorials/_index.md:29]: I dont like the sound of `Inter-Artifact`. should we just call is artifact dependency?
```suggestion
| [Artifact Dependencies]({{< relref "/docs/tutorials/artifact-dependencies" >}}) |
```

**Comment by gsquared94** [comment]
perhaps I should add a top level doc also.

File: docs/content/en/docs/tutorials/artifact-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+This artifact is used as the base image for the `leeroy-app` and `leeroy-web` artifacts. We express this dependency in the `skaffold.yaml` using the `requires` expression.
+{{% readfile file="samples/builders/artifact-dependencies/skaffold.yaml" %}}
+
+This allows us to use `BASE` as a build-arg in the Dockerfile for `leeroy-app` (and similarly for `leeroy-web`):
+ 
+{{% readfile file="samples/builders/artifact-dependencies/Dockerfile.app" %}}
+
+Skaffold orchestrates these builds in dependency order and injects the value of `BASE` with the image built for the `base` artifact while building `leeroy-web` and `leeroy-app`.
+You can run `skaffold build` to build these artifacts, or `skaffold dev` to build and deploy with file change monitoring.
```
Comment (tejal29) [inline docs/content/en/docs/tutorials/artifact-dependencies.md:32]: I would say, you can ask users to run `skaffold dev` and explain with output snapshot, 
1. how they could detect builds are built in order form the output.
2. demonstrate, how changing a base artifact, re-builds derived artifact. 

A brief explanation on `concurrency` field.

A tutorial, (most GCP) have format, 
1. Before you begin
2. <actions>
3. Cleanup - If we mention `skaffold dev` as a step, then just mention hitting cntrl C would be a good clean up or `skaffold delete`

**Review by tejal29 — CHANGES_REQUESTED** [review]

File: docs/content/en/docs/tutorials/_index.md (hunk 1 above)
Comment (gsquared94) [reply docs/content/en/docs/tutorials/_index.md]: done

File: docs/content/en/docs/tutorials/artifact-dependencies.md (hunk 2 above)
Comment (gsquared94) [reply docs/content/en/docs/tutorials/artifact-dependencies.md]: Done.

**Review by MarlonGamez — APPROVED** [review]
LGTM. I think the new tutorial is nice!

**Pull request merged**